### PR TITLE
initial support for ppc64 targets

### PR DESCRIPTION
**Pull request by q66**
This adds enough stuff so that Void can be bootstrapped (from source, including cross) for all new targets.

This supports:

- ppc64le (little endian elfv2 glibc)
- ppc64le-musl (little endian musl)
- ppc64-musl (big endian musl)

ELFv1 targets are not supported, musl uses ELFv2 on all endians by default. The cross toolchains for both musl targets are tested. ~~The cross toolchain for glibc is not, as I am myself running the glibc system and don't have another Linux installation to test with.~~ did some initial testing on another system, added fixes

~~Still pondering tuning the big endian binaries for POWER8+ (it should work without causing compatibility issues on older POWER as long as still targeting it)~~ edit: now always tuning for power9 and supporting ppc970 (POWER4/G5) and newer on big endian and POWER8 and newer on little endian, also always compiling with altivec and musl always uses 64 bit long doubles as is expected for it

I have plenty of other fixes lined up for various packages, but first the initial plumbing needs to be out of the way.

May/probably will conflict with https://github.com/void-linux/void-packages/pull/6108, so depending on which one makes it in first, either this or the other PR will have to be rebased again.

### CODE REVIEW

File: srcpkgs/base-files/INSTALL [hunk 1]
```diff
@@ -48,6 +48,10 @@ make_system_dirs() {
 		install -d usr/lib32
 		ln -sf usr/lib32 lib32
 	fi
+	if [ "${ARCH}" = "ppc64le" ]; then
+		ln -sf usr/lib lib64
+		cd usr && ln -sf lib lib64 && cd ..
```
Comment (CameronNemo) [inline srcpkgs/base-files/INSTALL]: Why can't you avoid the chdir here?

Comment (q66) [reply srcpkgs/base-files/INSTALL]: No particular reason. I just copied this bit from the x86_64 condition above.

**Comment by q66** [comment]
Okay, I think I will switch tunes for all targets. LE targets will add `-mtune=power9` (AFAIK, does not make an actual difference on GCC 8 but will be useful for later), `-mcpu=powerpc64le` stays (so that it works on POWER8 and newer CPUs). BE targets will follow the same settings as Adelie Linux and have `-mcpu=970 -mtune=power9` to allow for G5 and newer compatibility while still tuning for POWER9. Additionally, all musl targets will add `-mlong-double-64` so that it does not have to be added in individual packages that break on it, and all targets will get `-maltivec` as that is always available.

**Comment by q66** [comment]
Updated all the tunes/CPU requirements.

**Comment by q66** [comment]
fixes for ppc64le glibc cross-toolchain

**Comment by q66** [comment]
rebased

**Comment by Gottox** [comment]
I think we're good to go here. I'm merging this first and later #6108. :+1:

File: common/build-style/cmake.sh [hunk 2]
```diff
@@ -13,6 +13,8 @@ do_configure() {
 			aarch64*) _CMAKE_SYSTEM_PROCESSOR=aarch64 ;;
 			arm*) _CMAKE_SYSTEM_PROCESSOR=arm ;;
 			mips*) _CMAKE_SYSTEM_PROCESSOR=mips ;;
+			ppc64le*) _CMAKE_SYSTEM_PROCESSIR=ppc64le ;;
```
Comment (jnbr) [inline common/build-style/cmake.sh]: typo

Comment (Gottox) [reply common/build-style/cmake.sh]: fixed :+1:

**Pull request merged**